services/visual_recognition/v3.js

```diff
@@ -0,0 +1,362 @@
+
+  var imageType = require('image-type');
+  var url = require('url');
+  var temp = require('temp');
+  var fileType = require('file-type');
+  var fs = require('fs');
+  var async = require('async');
+
+  // temp is being used for file streaming to allow the file to arrive so it can be processed. 
+  temp.track();
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then the api key is stored in the variable s_apikey. 
+  //
+  // This separation between s_apikey and apikey is to allow 
+  // the end user to modify the key credentials when the service is not bound.
+  // Otherwise, once set apikey is never reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Taking this line out as codacy was complaining about it. 
+  // var services = cfenv.getAppEnv().services,
+
+  var apikey, s_apikey;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/visual recognition/i);
+
+  if (service) {
+    s_apikey = service.apikey;
+  }
+
+  RED.httpAdmin.get('/watson-visual-recognition/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  // Utility functions that check for image buffers, urls and stream data in
+ 
+  function imageCheck(data) {
+    return data instanceof Buffer && imageType(data) !== null;
+  };
+
+  function urlCheck(str) {
+    var parsed = url.parse(str)
+    return (!!parsed.hostname && !!parsed.protocol && str.indexOf(' ') < 0);
+  };
+
+  function stream_buffer(file, contents, cb) {
+    fs.writeFile(file, contents, function (err) {
+      if (err) throw err;
+      cb();
+    });
+  };
+
+  // Utility function that performs the Watson Visual Recognition call. 
+  // the cleanup removes the temp storage, and I am not sure whether 
+  // it should be called here or after alchemy returns and passed
+  // control back to cbdone.
+
+  function performAction(params, feature, cbdone) {
+
+        var visualRecognition = watson.visual_recognition({
+          api_key: apikey,
+          version: 'v3',
+          version_date: '2016-05-19'
+        });
+    switch(feature)
+    {
+      case 'classifyImage' : 
+        visualRecognition.classify(params, cbdone);
+        break;
+      case 'detectFaces':
+        visualRecognition.detectFaces(params, cbdone);
+        break;
+      case 'recognizeText':
+        visualRecognition.recognizeText(params, cbdone);
+        break;
+      case 'createClassifier':
+        visualRecognition.createClassifier(params, cbdone);
+        break;
+      case 'retrieveClassifiersList':
+        visualRecognition.listClassifiers(params, cbdone);
+        break;
+      case 'retrieveClassifierDetails':
+        visualRecognition.getClassifier(params, cbdone);
+        break;
+      case 'deleteClassifier':
+        visualRecognition.deleteClassifier(params, cbdone);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 6 space characters but found 8.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2690757886/issues/source?bid=3225115&fileBranchId=3339134#l117)